### DISCUSSION TITLE
context assert error "Assertions require every name in the call target to be declared with an explicit type annotation"

### DISCUSSION THREAD

**Discussion by nameOfUser0236** [oakserver/oak#446] [Q&A] [answered]
oak's context's assert function generates the following error:
 `Assertions require every name in the call target to be declared with an explicit type annotation`
as shown by this example:
``` typescript
import * as oak from "https://deno.land/x/oak/mod.ts";

const app = new oak.Application();

app.use((ctx) => {
    ctx.assert(true);
});

await app.listen({ port: 8000 });
```
I suspect that it is because the condition is any.
Some help would be appreciated as my current workaround is very painful.

**Comment by kitsonk** [comment, marked as answer]
This is a _design limitation_ of TypeScript (see: https://github.com/microsoft/TypeScript/issues/36931).

You need to do something like:

```ts
import * as oak from "https://deno.land/x/oak/mod.ts";
import type { Context } from "https://deno.land/x/oak/mod.ts";

const app = new oak.Application();

app.use((ctx: Context) => {
    ctx.assert(true);
});

await app.listen({ port: 8000 });
```

**Reply by nameOfUser0236** [reply]
Thank you!
